gitpod-io/gitpod
# [context parser] Check current ref for Docker image existence

## Description

Whenever we try to parse the context for a workspace, we check that any Dockerfile mentioned in the `.gitpod.yml`'s `image.file` actually exists and we also check when it was last modified.

The problem is that for most providers, we did the existence check by listing commits for the given path, which gave false-positives for Dockerfiles that once existed but no longer do on the tip of the git reference.

In addition, we now don't let this error block workspace starts. Instead, we attach a warning to the workspace metadata and show it in the UI when starting.

## Related Issue(s)

Follow-up for CLC-851 and https://github.com/gitpod-io/gitpod/pull/20267

## How to test

You can try parsing https://github.com/spring-projects/spring-petclinic on the [preview environment](https://ft-fix-cur67996acb2a.preview.gitpod-dev.com/workspaces). It should let you create a workspace, but with a warning.
